File: docs/剑指offer/Python/05-用两个栈实现队列.py
# -*- coding:utf-8 -*-
class Solution:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []
        
    def push(self, node):
        # write code here
        self.stack1.append(node)
        
    def pop(self):
        # return xx
        if len(self.stack1) == 0 and len(self.stack2) == 0:
            return 
        elif len(self.stack2) == 0:
            while len(self.stack1) > 0:
                self.stack2.append(self.stack1.pop())
           
        return self.stack2.pop()

File: docs/jianzhioffer/javascript/05-用两个栈实现队列.js
let stack1 = [];
let stack2 = [];
function push(node)
{
    // write code here
    stack1.push(node)
}
function pop()
{
    // write code here
    if(stack2.length === 0) {
        while (stack1.length!==0) {
            stack2.push(stack1.pop())
        }
    }
    return stack2.pop()
}
module.exports = {
    push : push,
    pop : pop
};
